components/common/src/main/java/com/hotels/styx/NettyExecutor.java

```diff
@@ -70,7 +72,11 @@ private NettyExecutor(EventLoopGroup eventLoopGroup,
     }
 
     public void shut() {
-        eventLoopGroup.shutdownGracefully();
+        try {
+            eventLoopGroup.shutdownGracefully(0, 0, TimeUnit.SECONDS).await(5000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
This is clearing the interrupted flag, we've discussed previously it might not be the best option.

Reply:
Thanks. A good catch :-)

Reply:
Fixed.